src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -290,6 +308,17 @@ private bool StartCore(ProcessStartInfo startInfo)
             return true;
         }
 
+        /// <summary>Gets arguments.</summary>
+        /// <param name="startInfo">The start info with which to start the process.</param>
+        private string[] GetArgsToOpenFile(ProcessStartInfo startInfo)
```

Comment:
>GetArgsToOpenFile(ProcessStartInfo startInfo) [](start = 25, length = 45)

I notice this can be called only when startInfo.UseShellExecute is true. if this is the case, you may not need this method and just use ParseArgv(startInfo, GetPathToOpenFile()); in teh place you are calling the method